resnet.py

```diff
@@ -0,0 +1,94 @@
+import json
+from pathlib import Path
+from typing import Any, Optional, Tuple
+
+import torch
+import torchvision.models as models
+from pytorch_lightning import LightningModule
+from torch import Tensor
+from torch.nn import BCEWithLogitsLoss, Linear
+from torch.optim import AdamW
+from torch.optim.lr_scheduler import CosineAnnealingLR
+from torchio.transforms import RescaleIntensity
+
+from config import Config
+
+
+class LitResnet(LightningModule):
+    def __init__(self, out_features: int, config: Config, resnet_version: int = 18) -> None:
+        super().__init__()
+        resnets = {
+            18: models.resnet18,
+            34: models.resnet34,
+            50: models.resnet50,
+            101: models.resnet101,
+            152: models.resnet152,
+        }
+        self.out_features = out_features
+        self.bce = BCEWithLogitsLoss()
+        self.resnet_version = resnet_version
+        self.resnet_model = resnets[resnet_version](pretrained=True)
+        linear_size = list(self.resnet_model.children())[-1].in_features
+        self.resnet_model.fc = Linear(linear_size, out_features)
+        self.config = config
+
+    def save_configs(self, log_dir: Path):
+        log_dir = Path(log_dir) / "config.json"
+        self.config.__dict__["resnet_version"] = self.resnet_version
+        json.dump(self.config.__dict__, open(log_dir, "w"))
```

Comment:
I am pretty sure you leak a file descriptor here, since you never `close` `log_dir`. Doesn't matter much here, but this is a bad thing in general. Just use a context manager to automate closing:
```suggestion
        with open(log_dir, "w") as handle:
            json.dump(self.config.__dict__, handle)
```

Reply:
Okay, done